### PR TITLE
add gem ratyrate

### PR DESCRIPTION
**Pull request by sonvc94**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -19,6 +19,7 @@ gem "paperclip", "~> 4.1"
 gem "sdoc", "0.4.0", group: :doc
 gem "ransack", "1.7"
 gem "kaminari"
+gem 'ratyrate'
```
Comment (tuannd255) [inline Gemfile]: double quote

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -12,5 +12,6 @@
 //
 //= require jquery
 //= require jquery_ujs
-//= require turbolinks
```
Comment (tuannd255) [inline app/assets/javascripts/application.js]: sao loai bo tubolinks vay @@

File: app/controllers/rater_controller.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class RaterController < ApplicationController
+
+  def create
+    if user_signed_in?
+      obj = params[:klass].classify.constantize.find params[:id]
```
Comment (tuannd255) [inline app/controllers/rater_controller.rb]: sao van con dung find a` @@

File: app/controllers/rater_controller.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+      obj.rate params[:score].to_f, current_user, params[:dimension]
+
+      render :json => true
```
Comment (tuannd255) [inline app/controllers/rater_controller.rb]: co the bo => khong?

File: app/models/average_cache.rb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+class AverageCache < ActiveRecord::Base
+  belongs_to :rater, :class_name => "User"
```
Comment (tuannd255) [inline app/models/average_cache.rb]: =>???

File: app/models/overall_average.rb [hunk 6]
```diff
@@ -0,0 +1,3 @@
+class OverallAverage < ActiveRecord::Base
+  belongs_to :rateable, :polymorphic => true
```
Comment (tuannd255) [inline app/models/overall_average.rb]: =>??

File: config/routes.rb [hunk 7]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  post '/rate' => 'rater#create', :as => 'rate'
```
Comment (tuannd255) [inline config/routes.rb]: double quote

File: app/controllers/rater_controller.rb [hunk 8]
```diff
@@ -0,0 +1,13 @@
+class RaterController < ApplicationController
+
+  def create
+    if user_signed_in?
+      obj = params[:klass].classify.constantize.find_by_id params[:id]
+      obj.rate params[:score].to_f, current_user, params[:dimension]
+
```
Comment (hatd) [inline app/controllers/rater_controller.rb]: cách ra làm gì

**Pull request merged**